AI #1785: Add Invoice Detail and Billing Period datasets (#1900)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/datasets/invoice_detail/columns/invoicedetailcreated.md

# Invoice Detail Created

Invoice Detail Created is the timestamp when the [Invoice Detail](#datasets.invoicedetail) record was first created. This timestamp facilitates auditability of the charge and invoice lifecycle, allowing the FinOps practitioner to distinguish between the time of service consumption and the time of financial record generation.

## Requirements

InvoiceDetailCreated adheres to the following requirements:

* InvoiceDetailCreated MUST be of type Date/Time.
* InvoiceDetailCreated MUST conform to [DateTimeFormat](#attributes.date/timeformat) requirements.

> * InvoiceDetailCreated MUST represent the moment in time the Invoice Detail record was instantiated.

Reviewer: In the Invoice Handling attribute (addressed in PR #1501), the following requirement specific to the InvoiceDetailCreated column is currently listed.

As mentioned in [this comment](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1900/files#r2812889585), we are still discussing the criteria for attribute-level requirements vs. column-level requirements, and are considering relocating (and rephrasing) this requirement from the attribute to the column itself.

> * [InvoiceDetail.InvoiceDetailCreated](#datasets.invoicedetail.invoicedetailcreated) for the given InvoiceDetail.BillingPeriodStart and InvoiceDetail.InvoiceIssuerName MUST be earlier than or equal to [BillingPeriod.BillingPeriodLastUpdated](#datasets.billingperiod.billingperiodlastupdated) for the same BillingPeriod.BillingPeriodStart and BillingPeriod.InvoiceIssuerName when [BillingPeriod.BillingPeriodStatus](#datasets.billingperiod.billingperiodstatus) is "Closed".

Author: @ijurica Understood.  Let's meet to decide the right wording to use here.

Author: @ijurica 
I have addressed this.  Please review and resolve.

Reviewer: LGTM. Resolving

# Invoice Detail Created

> Invoice Detail Created is the timestamp when the [Invoice Detail](#datasets.invoicedetail) record was first created. This timestamp facilitates auditiability of the charge and invoice lifecycle, allowing the FinOps practitioner to distinguish betwee the time between the time of service consumption and the time of financial record generation.

Reviewer 2: ```suggestion
Invoice Detail Created is the timestamp when the [Invoice Detail](#datasets.invoicedetail) record was first created. This timestamp facilitates auditability of the charge and invoice lifecycle, allowing the FinOps practitioner to distinguish between the time of service consumption and the time of financial record generation.
```

## Requirements

> InvoiceDetailCreated adheres to the following requirements:

Reviewer 3: ```suggestion
InvoiceDetailCreated MUST adhere to the following requirements:
```

* InvoiceDetailCreated MUST be of type Date/Time.

> * InvoiceDetailCreated MUST conform to [DateTimeFormat](#attributes.date/timeformat) requirements.

Reviewer: ```suggestion
* InvoiceDetailCreated MUST conform to [DateTimeFormat](#attributes.date/timeformat) requirements.
* InvoiceDetailCreated MUST NOT be null.
```
